PowerArubaCX/Public/Tacacs.ps1

```diff
@@ -0,0 +1,375 @@
+
+        $_tacacs | add-member -name "group" -membertype NoteProperty -Value $_group
+
+        $_tacacs | add-member -name "auth_type" -membertype NoteProperty -Value $auth_type
+
+        if ( $PsBoundParameters.ContainsKey('passkey') ) {
+            $_tacacs | add-member -name "passkey" -membertype NoteProperty -Value $passkey
+        }
+
+        if ( $PsBoundParameters.ContainsKey('timeout') ) {
+            $_tacacs | add-member -name "timeout" -membertype NoteProperty -Value $timeout
+        }
+
+        if ( $PsBoundParameters.ContainsKey('user_group_priority') ) {
+            $_tacacs | add-member -name "user_group_priority" -membertype NoteProperty -Value $user_group_priority
+        }
+
+        if ( $PsBoundParameters.ContainsKey('tracking_enable') ) {
+            if ($tracking_enable) {
+                $_tacacs | add-member -name "tracking_enable" -membertype NoteProperty -Value $true
+            }
+            else {
+                $_tacacs | add-member -name "tracking_enable" -membertype NoteProperty -Value $false
+            }
+        }
+
+        $response = Invoke-ArubaCXRestMethod -uri $uri -method 'POST' -body $_tacacs -connection $connection
+        $response
+
+        Get-ArubaCXTacacsServer -address $address -port $port -vrf $vrf
+
+    }
+
+    End {
+    }
+}
+
+function Get-ArubaCXTacacsServer {
+
+    <#
+        .SYNOPSIS
+        Get list of TACACS server configured
+
+        .DESCRIPTION
+        Get list of TACACS server configured (ip, group, port...)
+
+        .EXAMPLE
+        Get-ArubaCXTacacsServer -vrf default
+
+        Get list of TACACS server configured (ip, group, port...) on default vrf
+
+        .EXAMPLE
+        Get-ArubaCXTacacsServer -address 192.2.0.1 -port 49
+
+        Get TACACS server with ip 192.2.0.1 and port 49
+    #>
+
+    [CmdletBinding(DefaultParametersetname = "Default")]
+    Param(
+        [Parameter (Mandatory = $true, ParameterSetName = "address")]
+        [ipaddress]$address,
+        [Parameter (Mandatory = $true, ParameterSetName = "address")]
+        [int]$port = 49,
+        [Parameter (Mandatory = $false)]
+        [string]$vrf = "default",
+        [Parameter(Mandatory = $false)]
+        [ValidateRange(1, 4)]
+        [Int]$depth,
+        [Parameter(Mandatory = $false, ParameterSetName = "address")]
+        [ValidateSet("configuration", "status", "statistics", "writable")]
+        [String]$selector,
+        [Parameter(Mandatory = $false)]
+        [String[]]$attributes,
+        [Parameter(Mandatory = $false)]
+        [switch]$vsx_peer,
+        [Parameter (Mandatory = $False)]
+        [ValidateNotNullOrEmpty()]
+        [PSObject]$connection = $DefaultArubaCXConnection
+    )
+
+    Begin {
+    }
+
+    Process {
+
+
+        $invokeParams = @{ }
+        if ( $PsBoundParameters.ContainsKey('depth') ) {
+            $invokeParams.add( 'depth', $depth )
+        }
+        else {
+            #by default set depth to 2 to show items
```

Comment:
???

Reply:
If depth is not set to 2, we don't see all attributes of TACACS servers

Reply:
it is by design..